components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/common/mappings/PublisherCommonUtils.java

```diff
@@ -709,6 +709,7 @@ public static API addAPIWithGeneratedSwaggerDefinition(APIDTO apiDto, String oas
             String apiDefinition = oasParser.generateAPIDefinition(swaggerData);
             apiToAdd.setSwaggerDefinition(apiDefinition);
         }
+        apiToAdd.setOrganizationId(organizationId);
```

Comment:
shall we do this inside `prepareToCreateAPIByDTO` method itself?

Reply:
Fixed https://github.com/wso2/carbon-apimgt/pull/9684/commits/cc8517b20f7dbec93569cf9ab02809b6b9148cbe